### PR TITLE
feat: Accessibility of Alert close button

### LINKED ISSUE #18736: Accessibility of Alert close button
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/alert/](https://ant.design/components/alert/)

### Steps to reproduce
1) Navigate with the TAB key to any Alert close button.
2) Try to click Enter or Space.

### What is expected?
Alert is being closed.

### What is actually happening?
Since the close button is not actually a `<button>` element, it doesn't get key bindings automatically making it unaccessible by blind users or users with motoric issues.

| Environment | Info |
|---|---|
| antd | 3.23.2 |
| React | Irrespectively |
| System | Irrespectively |
| Browser | Irrespectively |

---
The best way is to replace the `<span>` with `<button>` and style it accordingly. This will offer the most consistent behavior across browsers and screen readers.
Another (unpreffered) solution is to add manual key bindings (Enter, Space) to it that will trigger the onClick event.

### PR DESCRIPTION
**Pull request by MrHeer**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Accessibility

### 🔗 Related issue link
close #18736

### 💡 Background and solution
replace the `<span>` with `<button>` and style it accordingly

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   :wheelchair: Accessibility of `Alert` close button        |
| 🇨🇳 Chinese |     :wheelchair:  `Alert` 关闭按钮的可访问性      |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by orzyyyy — APPROVED** [review]

**Comment by yoyo837** [comment]
`:focus` diffs 
image
image

If diffs is accepted, then this is all.

**Comment by afc163** [comment]
It is ugly, I don't think our designer will accept this.

**Comment by afc163** [comment]
Maybe a darker color would be ok.

**Comment by MrHeer** [comment]
I didn't see any difference

Screen record from 2019-09-10 11 12 26

**Comment by orzyyyy** [comment]
test
Difference does exist, it's my carelessness.

File: components/alert/style/index.less [hunk 1]
```diff
@@ -74,8 +74,15 @@
     overflow: hidden;
     font-size: @font-size-sm;
     line-height: 22px;
+    border: none;
+    background-color: transparent;
     cursor: pointer;
 
+    &:focus {
```
Comment (orzyyyy) [inline components/alert/style/index.less]: Maybe it's better to remove `:focus`, it looks like before.
test2

**Comment by MrHeer** [comment]
I found it is different in Chrome, but it is same in Firefox.

**Comment by orzyyyy** [comment]
I prefer to use the default behavior, which matches the design and can save both trouble and your time.

File: components/alert/style/index.less (hunk 1 above)
Comment (yoyo837) [reply components/alert/style/index.less]: I agree with @orzyyyy .

**Review by yoyo837 — CHANGES_REQUESTED** [review]
remove `:focus`

**Comment by MrHeer** [comment]
> remove `:focus`

It will cause no effect in firefox

**Comment by MrHeer** [comment]
Sorry, I am going to the interview now, and I am dealing with it later.

**Review by afc163 — APPROVED** [review]

**Pull request merged**